tests/broadcast.rs

```diff
@@ -68,37 +63,43 @@ enum MessageScheduler {
 
 impl MessageScheduler {
     /// Chooses a node to be the next one to handle a message.
-    fn pick_node(&self, nodes: &BTreeMap<NodeId, TestNode>) -> NodeId {
+    fn pick_node<D: DistAlgorithm>(&self, nodes: &BTreeMap<D::NodeUid, TestNode<D>>) -> D::NodeUid {
```

Comment:
This is great. The scheduler looks generic enough to be made into a separate module. I wonder if that makes sense at the moment.

Reply:
Nevermind, that would require abstracting over `TestNode` and that goes beyond the scope of this PR.

Reply:
Yes, let's move it to something like `tests/network` when we use it for other tests. (I didn't do it in this PR, to keep the diff readable.)